server/implementation/src/main/java/io/smallrye/graphql/bootstrap/Bootstrap.java

```diff
@@ -409,28 +418,53 @@ private void addGroupedRootObject(GraphQLObjectType.Builder rootBuilder,
             GraphQLFieldDefinition namedField = graphQLFieldDefinitionBuilder.build();
 
             this.codeRegistryBuilder.dataFetcherIfAbsent(
-                    FieldCoordinates.coordinates(rootName, namedField.getName()),
+                    FieldCoordinates.coordinates(rootName + suffix, namedField.getName()),
                     dummyDataFetcher);
-
             rootBuilder.field(namedField);
+
+            return List.of(namedField);
         }
+        return List.of();
+    }
+
+    private List<GraphQLFieldDefinition> getGraphQLFieldDefinition(GroupContainer groupContainer, String rootName,
+            String suffix) {
+        String name = prettyString(groupContainer.getName());
+        String namedTypeName = rootName + name + suffix;
+
+        GraphQLObjectType.Builder wrapperBuilder = GraphQLObjectType.newObject()
+                .name(namedTypeName)
+                .description(groupContainer.getDescription());
+
+        return groupContainer
+                .getContainer()
+                .values()
+                .stream()
+                .map(namespace -> addNamespacedRootObject(
+                        wrapperBuilder, namespace, rootName + prettyString(groupContainer.getName()), suffix))
+                .flatMap(Collection::stream)
+                .collect(Collectors.toList());
     }
 
-    private GraphQLObjectType createNamedType(String parent, Group group, Set<Operation> operations) {
-        String namedTypeName = group.getName() + parent;
+    private GraphQLObjectType createNamespaceType(String root, String suffix, GroupContainer namespace,
+            Set<Operation> operations, List<GraphQLFieldDefinition> graphQLFieldDefinitions) {
+        String name = prettyString(namespace.getName());
+
+        String namedTypeName = root + name + suffix;
         GraphQLObjectType.Builder objectTypeBuilder = GraphQLObjectType.newObject()
                 .name(namedTypeName)
-                .description(group.getDescription());
+                .description(namespace.getDescription());
 
         // Operations
         for (Operation operation : operations) {
             operation = eventEmitter.fireCreateOperation(operation);
 
-            GraphQLFieldDefinition graphQLFieldDefinition = createGraphQLFieldDefinitionFromOperation(namedTypeName,
-                    operation);
+            GraphQLFieldDefinition graphQLFieldDefinition = createGraphQLFieldDefinitionFromOperation(
+                    namedTypeName, operation);
             objectTypeBuilder = objectTypeBuilder.field(graphQLFieldDefinition);
         }
 
+        objectTypeBuilder.fields(graphQLFieldDefinitions);
```

Comment:
isn't the second one unnecessary?

Reply:
here passed nested objects. 
`@Namespace("admin")`, `@Namespace("admin/users")`. Each contains some operations. Here builds admin definition (with its opeartions), and in the next adds definitions 'users' and etc..